### DISCUSSION TITLE
Rail signal overlay

### DISCUSSION THREAD

**Discussion by the-moonwitch** [voxeltycoon/issues#380] [UI&UX] [answered]
**Is your feature request related to a problem? Please describe.**
Signals are a big point of confusion in the community. I personally have given up on using trains because I could never understand how they work. 

**Describe the solution you'd like**
Railway Empire, a favorite game of mine, has a very simple overlay that shows the effect of signals on train pathing:
![railwayempire](https://user-images.githubusercontent.com/35173529/60719670-b70b7380-9f28-11e9-837e-2f308ef898cb.gif)

Something similar for Voxel Tycoon would be great, an overlay that displays signal blocks to be triggered eg. while placing signals.

**Describe alternatives you've considered**
A clear in-game explanation of signal mechanics would be useful, but I think the overlay would be a great aid for clarity and the train learning curve.

**Comment by katepolak** [comment]
Similar idea already suggested in #30 (link is invalid because of conversions to different trackers), marking as duplicate but not closing as the suggestions aren't exactly identical.

**Comment by the-moonwitch** [comment]
Apologies for not being clear; #30 is asking for an overlay that would provide information that is already visible (though perhaps not as clearly) to the player - the signal lights.

I meant something more akin to the block diagrams that have been showing up on the discord and subreddit, examples:
![image](https://user-images.githubusercontent.com/35173529/60738650-606d5c00-9f5f-11e9-9306-845e0cb24d8a.png)
![image](https://user-images.githubusercontent.com/35173529/60738654-62cfb600-9f5f-11e9-97c5-2f724ee83970.png)

An overlay to help demystify how signals work, rather than provide existing information more clearly (as in #30)

**Comment by katepolak** [comment]
Yes, an overlay like this is definitely planned.

**Comment by kristofeI** [comment]
My suggestion comes from the same sentiment:
**Signals are a big point of confusion in the community.**
Although descriptions i-game are rich and detailed (also in Polish version), I understood the signals (but I'm also not 100% sure) only after I read wiki article. Maybe add some image examples in-game? I'm not sure, but signals descriptions certainly need more clarification.

**Comment by victorleblais** [comment]
When building signals you now have arrows indicating the signal's status:
![image](https://user-images.githubusercontent.com/4365075/115538015-b8dc2d80-a29b-11eb-9449-d05d4270b0be.png)

Signal block visualisation is something that is planned on the roadmap:
https://voxeltycoon.xyz/roadmap
![image](https://user-images.githubusercontent.com/4365075/115538088-cf828480-a29b-11eb-9ac9-4c6ffd4d5541.png)

**Comment by victorleblais** [comment, marked as answer]
It is now implemented in the experimental branch
![image](https://user-images.githubusercontent.com/4365075/117546028-a64a4e00-b028-11eb-9aa1-d4c659d871fa.png)
